### PR TITLE
Adding primary wrangler and wrangling status in project list

### PR DESCRIPTION
**Pull request by aaclan-ebi**
This PR needs changes in core: https://github.com/ebi-ait/ingest-core/pull/70

**Changes:**

For ticket ebi-ait/dcp-ingest-central#313 :
- Adding primary wrangler and wrangling status in project list

For ticket ebi-ait/dcp-ingest-central#317 :
- Applied Enrique and Gab's feedback

**Screenshots:**
Screenshot 2021-05-28 at 18 19 05
Screenshot 2021-05-28 at 18 18 45

Screenshot 2021-05-28 at 18 18 38

### CODE REVIEW

File: client/src/app/shared/components/project-list/project-list.component.ts [hunk 1]
```diff
@@ -61,6 +67,7 @@ export class ProjectListComponent implements OnInit {
       [ProjectColumn.short_name]: 'HCA Project ID',
       [ProjectColumn.project_title]: 'Project Title',
       [ProjectColumn.last_updated]: 'Last Updated',
+      [ProjectColumn.wrangling_status]: 'Wrangling Status',
       [ProjectColumn.primary_contributor]: 'Primary Contributor',
       [ProjectColumn.primary_wrangler]: 'Primary Wrangler'
```
Comment (MightyAx) [inline client/src/app/shared/components/project-list/project-list.component.ts:72]: I've checked that these columns don't show if the user is not a wrangler...
But I can't see the code that hides them, could you point me in the right direction?

**Review by MightyAx — COMMENTED** [review]
Do we need to worry about this errors?
Screenshot 2021-06-01 at 11 55 29

File: client/src/app/all-projects/all-projects.component.ts [hunk 2]
```diff
@@ -23,7 +20,9 @@ export class AllProjectsComponent implements OnInit, OnDestroy {
     ProjectColumn.short_name,
     ProjectColumn.project_title,
     ProjectColumn.primary_contributor,
-    ProjectColumn.last_updated
+    ProjectColumn.last_updated,
+    ProjectColumn.primary_wrangler,
+    ProjectColumn.wrangling_status
```
Comment (jacobwindsor) [inline client/src/app/all-projects/all-projects.component.ts:25]: Should be `wrangling_state` right. Also camelCase is preferred in JS

File: client/src/app/my-projects/my-projects.component.ts [hunk 3]
```diff
@@ -47,7 +47,9 @@ export class MyProjectsComponent implements OnInit, OnDestroy {
             ProjectColumn.short_name,
             ProjectColumn.project_title,
             ProjectColumn.primary_contributor,
-            ProjectColumn.last_updated
+            ProjectColumn.last_updated,
+            ProjectColumn.primary_wrangler,
```
Comment (jacobwindsor) [inline client/src/app/my-projects/my-projects.component.ts:51]: `wrangling_state` here too

**Review by jacobwindsor — CHANGES_REQUESTED** [review]

File: client/src/app/shared/components/project-list/project-list.component.ts (hunk 1 above)
Comment (aaclan-ebi) [reply client/src/app/shared/components/project-list/project-list.component.ts:72]: I've only added them in all-projects component https://github.com/ebi-ait/ingest-ui/pull/94/files#diff-68fc70737bc6a261cf75d759bd0a9ecdef83917b60b1c91c78ebb300bf227d7cR24-R25

Comment (aaclan-ebi) [reply client/src/app/shared/components/project-list/project-list.component.ts:72]: Yup, sorry i didnt see the errors, thanks for checking! fixing them now!

Comment (aaclan-ebi) [reply client/src/app/shared/components/project-list/project-list.component.ts]: You could specify the columns to include when you use the project-list component.
The additional columns are only added in the all-projects and not in the my-projects component.

File: client/src/app/all-projects/all-projects.component.ts (hunk 2 above)
Comment (aaclan-ebi) [reply client/src/app/all-projects/all-projects.component.ts]: yup, thanks! i've renamed them!

File: client/src/app/shared/components/project-list/project-list.component.ts (hunk 1 above)
Comment (MightyAx) [reply client/src/app/shared/components/project-list/project-list.component.ts]: Found the choice is made here: https://github.com/ebi-ait/ingest-ui/blob/fa031d621c1df39867259a9d011a4d2abec8141a/client/src/app/my-projects/my-projects.component.ts#L43

Adding in context for prosperity.

File: client/src/app/shared/components/project-list/project-list.component.html [hunk 4]
```diff
@@ -1,4 +1,4 @@
-<table class="vf-table vf-table--striped vf-table--bordered white-border" *ngIf="projects && wranglers">
+<table class="vf-table vf-table--striped vf-table--bordered white-border" *ngIf="projects && (!isWrangler || (isWrangler && wranglers))">
```
Comment (MightyAx) [inline client/src/app/shared/components/project-list/project-list.component.html:1]: can be logically rewritten:
```
*ngIf="projects && (!isWrangler || wranglers)
```

File: client/src/app/shared/components/project-list/project-list.component.ts [hunk 5]
```diff
@@ -6,60 +6,47 @@ import {AuthService} from '../../services/auth.service';
...
-    return this._columns;
-  }
+  @Input()
+  columns: ProjectColumn[];
 
   @Input()
-  set columns(columns: ProjectColumn[]) {
-    this._columns = columns;
-  }
+  showUnassignedActions: Boolean;
```
Comment (MightyAx) [inline client/src/app/shared/components/project-list/project-list.component.ts:27]: This input is no longer used anywhere in the component, should be removed.

**Review by MightyAx — CHANGES_REQUESTED** [review]
I'll make a commit to resolve the following.

File: client/src/app/shared/components/project-list/project-list.component.html (hunk 4 above)
Comment (aaclan-ebi) [reply client/src/app/shared/components/project-list/project-list.component.html:1]: ah thank you!

**Review by MightyAx — APPROVED** [review]

**Review by jacobwindsor — APPROVED** [review]

**Pull request merged**